qmf/src/plugins/messageservices/imap/imapservice.cpp

```diff
@@ -1763,18 +1752,30 @@ bool ImapService::cancelOperation(QMailServiceAction::Status::ErrorCode code, co
 
 void ImapService::restartPushEmail()
 {
+    qMailLog(Messaging) << "Attempting to restart push email for account" << _accountId
+                        << QMailAccount(_accountId).name();
     cancelOperation(QMailServiceAction::Status::ErrInternalStateReset, tr("Initiating push email"));
-    initiatePushEmail();
+    if (_networkSession && _networkSession->state() == QNetworkSession::Connected) {
```

Comment:
This test/branch is duplicated in the function below, with the exception that some state is not cleared in this case.  This seems subtle and comlpex; unless that's really necessary, I think it would be better to just call initiatePushEmail() here.